src/App.tsx

```diff
@@ -23,201 +211,108 @@ export const App: React.FC = () => {
-              All
-            </a>
-
-            <a
-              data-cy="FilterLinkActive"
-              href="#/active"
-              className="filter__link"
-            >
-              Active
-            </a>
-            <a
-              data-cy="FilterLinkCompleted"
-              href="#/completed"
-              className="filter__link"
-            >
-              Completed
-            </a>
-          </nav>
+            <footer className="todoapp__footer" data-cy="Footer">
+              <span className="todo-count" data-cy="todosCounter">
+                {`${todosLength()} items left`}
+              </span>
 
-          <button
-            data-cy="ClearCompletedButton"
-            type="button"
-            className="todoapp__clear-completed"
-          >
-            Clear completed
-          </button>
-        </footer>
-      </div>
+              <nav className="filter" data-cy="Filter">
+                <a
+                  data-cy="FilterLinkAll"
+                  href="#/"
+                  className={cn('filter__link',
+                    { selected: selectedFilter === FilterBy.ALL })}
+                  onClick={() => handleFilter(FilterBy.ALL)}
+                >
+                  All
+                </a>
 
-      <div
-        data-cy="ErrorNotification"
-        className="notification is-danger is-light has-text-weight-normal"
-      >
-        <button
-          data-cy="HideErrorButton"
-          type="button"
-          className="delete"
-        />
+                <a
+                  data-cy="FilterLinkActive"
+                  href="#/active"
+                  className={cn('filter__link',
+                    { selected: selectedFilter === FilterBy.ACTIVE })}
+                  onClick={() => handleFilter(FilterBy.ACTIVE)}
+                >
+                  Active
+                </a>
+                <a
+                  data-cy="FilterLinkCompleted"
+                  href="#/completed"
+                  className={cn('filter__link',
+                    { selected: selectedFilter === FilterBy.COMPLETED })}
+                  onClick={() => handleFilter(FilterBy.COMPLETED)}
+                >
+                  Completed
+                </a>
+              </nav>
 
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
-        <br />
-        Unable to update a todo
+              <button
+                data-cy="ClearCompletedButton"
+                type="button"
+                className={cn('todoapp__clear-completed', {
+                  hidden: todosCompletedLength() === 0,
+                })}
+                disabled={todosLength() === 0}
+                onClick={handleClearCompleted}
+              >
+                Clear completed
+              </button>
+            </footer>
+          </>
+        )}
       </div>
+
+      {(emptyTitleError
+        || addingError
+        || deletingError
+        || updatingError) && (
```

Comment:
let's create separate variable for storing the final result of this condition and use this variable in jsx